### PR TITLE
ConfigurableStreamMemory

### PR DESCRIPTION
**Pull request by jhonatanamado**
Set the memory of streams configurable via addRepackConfig

### CODE REVIEW

**Comment by drkovalskyi** [comment]
@jhonatanamado I don't think we need setConfStreamMemory. Function `addRepackConfig` requires to specify the stream where configuration is applied. In other words we set the default with `addRepackConfig(tier0Config, "Default", ...)` and modify configuration for a specific stream with extra calls of `addRepackConfig` specifying the stream name explicitly. Check how we use `addDataset`.

Please clean up the pull request no to have changes where they are not needed, i.e. white spaces, new lines etc.

File: src/python/T0/RunConfig/Tier0Config.py [hunk 1]
```diff
@@ -789,6 +789,15 @@ def addRepackConfig(config, streamName, **options):
     else:
         streamConfig.Repack.BlockCloseDelay = options.get("blockCloseDelay", 24 * 3600)
 
+    if hasattr(streamConfig.Repack, "SetStreamMemory"):
+        streamConfig.Repack.SetStreamMemory = options.get("setStreamMemory", streamConfig.Repack.SetStreamMemory)
+    else:
+        streamConfig.Repack.SetStreamMemory = options.get("setStreamMemory", 2000)
+
+    if hasattr(streamConfig.Repack, "SetConfStreamMemory"):
+        streamConfig.Repack.SetConfStreamMemory = options.get("setConfStreamMemory", streamConfig.Repack.SetConfStreamMemory)
```
Comment (germanfgv) [inline src/python/T0/RunConfig/Tier0Config.py:798]: As Dima already commented, this attribute is not necessary. We don't need a list of streams to modify, as what we will do is add a new configuration for each stream that needs a configuration different from the default.

File: src/python/T0/RunConfig/Tier0Config.py [hunk 2]
```diff
@@ -789,6 +789,15 @@ def addRepackConfig(config, streamName, **options):
     else:
         streamConfig.Repack.BlockCloseDelay = options.get("blockCloseDelay", 24 * 3600)
 
+    if hasattr(streamConfig.Repack, "SetStreamMemory"):
```
Comment (germanfgv) [inline src/python/T0/RunConfig/Tier0Config.py:792]: I would change the name of this attribute, as this name suggest an action (reads like a method's name). I would use something like `"MaxMemory"` instead.

**Review by germanfgv — COMMENTED** [review]
This is a lot closer to what we need, but it does need a few changes.

**Pull request closed without merging**